### ISSUE TITLE
http_server.sync_response.headers is case sensitive

### ISSUE THREAD

**Issue by mannharleen** [Jeffail/benthos#976]
Type = Minor Bug

The header field names are case insensitive ref: https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
However, `http_server.sync_response.headers` seems to be case sensitive i.e.
While, 
```
input:
  label: "in_http_server"
  http_server:
    path: /post
    sync_response: 
      headers: 
        Content-Type: "application/json"
```
works, 
```
input:
  label: "in_http_server"
  http_server:
    path: /post
    sync_response: 
      headers: 
        content-type: "application/json"
```
**doesn't (and it should!).**

**Comment by Jeffail** [comment]
Thanks for raising this @mannharleen, fixed in: https://github.com/Jeffail/benthos/commit/9ed999621a7750acede5e3150ce0e0a0da568a5c

**Comment by mannharleen** [comment]
> Thanks for raising this @mannharleen, fixed in: [9ed9996](https://github.com/Jeffail/benthos/commit/9ed999621a7750acede5e3150ce0e0a0da568a5c)

Cheers!
